src/main/java/uk/gov/companieshouse/company/links/service/AddExemptionsClient.java

```diff
@@ -53,6 +53,10 @@ public void patchLink(PatchLinkRequest linkRequest) {
             } else if (ex.getStatusCode() == 404) {
                 logger.info("HTTP 404 Not Found returned; "
                         + "company profile does not exist", DataMapHolder.getLogMap());
+                throw new RetryableErrorException(
```

Comment:
SonarQube is flagging these as duplicate code and failing. A solution could be to add another constructor that takes arguments that builds the message. Something like this: 
```
public RetryableErrorException(String companyNumber, String linkType, Exception exception) {
    super(String.format("Company profile [%s] does not exist when processing add %s link request",
            companyNumber, linkType), exception);
}
```

Reply:
done